scr/main/java/bittercode/util/DBUtil.java

```diff
@@ -0,0 +1,39 @@
+package com.bittercode.util;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.model.StoreException;
+
+public class DBUtil {
+
+    private static Connection connection;
+
+    static {
+
+        try {
+
+            Class.forName(DatabaseConfig.DRIVER_NAME);
+            
+            connection = DriverManager.getConnection(DatabaseConfig.CONNECTION_STRING, DatabaseConfig.DB_USER_NAME,
+                    DatabaseConfig.DB_PASSWORD);
+        } catch (SQLException | ClassNotFoundException e) {
+
+            e.printStackTrace();
+
+        }
+
+    }// End of static block
+
+    public static Connection getConnection() throws StoreException {
+
+        if (connection == null) {
+            throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE);
+        }
+
+        return null;

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>No connection retry in getConnection method</b></div>
<div id="fix">

The `getConnection()` method throws a `StoreException` if the `connection` is `null`, but doesn't attempt to re-establish the connection. Consider adding logic to retry connecting to the database before throwing the exception.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -30,7 +30,17 @@
      public static Connection getConnection() throws StoreException {
 
          if (connection == null) {
 -            throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE);
 +            try {
 +                Class.forName(DatabaseConfig.DRIVER_NAME);
 +                connection = DriverManager.getConnection(DatabaseConfig.CONNECTION_STRING, DatabaseConfig.DB_USER_NAME,
 +                        DatabaseConfig.DB_PASSWORD);
 +            } catch (SQLException | ClassNotFoundException e) {
 +                e.printStackTrace();
 +                throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE, e);
 +            }
 +        }
 +        if (connection == null) {
 +            throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE);
          }
 
 -        return null;
 +        return connection;
      }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/3#issuecomment-2434587212>#30653c</a></i></small>
</div>


---
🚩 Problem result:
- [ ] This issue was flagged incorrectly